utils/firebaseUtils.ts

```diff
@@ -16,12 +20,45 @@ export const getSubData = async (
   subpath: string,
   userId: string,
 ) => {
-  return await database()
-    .ref(path)
-    .child(userId + `/${subpath}`)
-    .once('value');
+  return await database().ref(path).child(`${userId}/${subpath}`).once('value');
+};
+
+export const getSubArray = async (
+  path: string,
+  subpath: string,
+  userId: string,
+) => {
+  const snapshot = await getSubData(path, subpath, userId);
+
+  const array: Array<ExerciseDetailsForm> = [];
+  snapshot.forEach((childSnapshot: FirebaseDatabaseTypes.DataSnapshot) => {
+    const item = childSnapshot.val();
+    const newItem = {
+      id: childSnapshot.key,
+      ...item,
+    };
+    array.push(newItem);
+    return undefined;
+  });
+
+  return array;
 };
 
 export const saveData = async <T>(path: string, userId: string, data: T) => {
   return await database().ref(path).child(userId).set(data);
 };
+
+export const pushUpdateItemInArray = async <T>(
```

Comment:
The name `pushUpdateItemInArray` seems to logically specific and not context specific, this functionality is in `firebaseUtils`, you can improve the name by adding that information to the name. Ex: `upsertFirebaseItem` or `upsertFirebaseItemByUserId` because you are specifically passing the user id and without it this wouldn't work.

Keep in mind that, you are passing 2 params to build 1 variable, which seems redundant.
```tsx
pushUpdateItemInArray({
userId: "123",
subpath: "test"
})
```
is the same as 
```tsx
pushUpdateItemInArray({
subpath: "123/test"
})
```

You are purposefully destroying the reusability by requiring `userId` to be passed